### DISCUSSION TITLE
Serve HTTPS + WSS requets, listening URL ?

### DISCUSSION THREAD

**Discussion by dev31337** [cesanta/mongoose#2405] [Ask questions, get answers] [answered]
Hi,

What is the right way to start a listener which can serve HTTPS requests but also when needed can upgrade to WSS ?

Should you mg_http_listen() "https://0.0.0.0:8443" or "wss://0.0.0.0:8443" ?

Or it does not matter, both are fine ?

Regards.

**Comment by cpq** [comment, marked as answer]
Both are fine.
The main thing is that on MG_EV_ACCEPT, the code calls mg_tls_init()
Then, upgrade on MG_EV_HTTP_MSG when required - see [example](https://github.com/cesanta/mongoose/blob/master/examples/websocket-server/main.c)
